components/eventing-controller/controllers/subscription/nats/unit_test.go

```diff
@@ -457,76 +458,45 @@ func setupTestEnvironment(t *testing.T) *TestEnvironment {
 	}
 }
 
-func createFakeClient(g *GomegaWithT) client.WithWatch {
+func createFakeClient(t *testing.T) client.WithWatch {
 	testEnv = &envtest.Environment{
 		CRDDirectoryPaths: []string{
 			filepath.Join("../../", "config", "crd", "bases"),
 			filepath.Join("../../", "config", "crd", "external"),
 		},
 	}
 	err := eventingv1alpha1.AddToScheme(scheme.Scheme)
-	g.Expect(err).Should(BeNil())
+	require.NoError(t, err)
 	return fake.NewClientBuilder().WithScheme(scheme.Scheme).Build()
 }
 
-func NewSubscription(options ...func(subscription *eventingv1alpha1.Subscription)) *eventingv1alpha1.Subscription {
-	sub := &eventingv1alpha1.Subscription{
-		ObjectMeta: v1.ObjectMeta{
-			Name:      "test",
-			Namespace: namespaceName,
-		},
-	}
-	for _, o := range options {
-		o(sub)
-	}
-	return sub
-}
-
-type SubscriptionOpt func(subscription *eventingv1alpha1.Subscription)
-
-func WithSink(sink string) SubscriptionOpt {
-	return func(sub *eventingv1alpha1.Subscription) {
-		sub.Spec.Sink = sink
-	}
-}
-func WithConditions(conditions []eventingv1alpha1.Condition) SubscriptionOpt {
-	return func(sub *eventingv1alpha1.Subscription) {
-		sub.Status.Conditions = conditions
-	}
-}
-func WithStatus(status bool) SubscriptionOpt {
-	return func(sub *eventingv1alpha1.Subscription) {
-		sub.Status.Ready = status
-	}
-}
-func WithFinalizers(finalizers []string) SubscriptionOpt {
-	return func(sub *eventingv1alpha1.Subscription) {
-		sub.ObjectMeta.Finalizers = finalizers
-	}
+// NewTestSubscription creates a test subscription
+func NewTestSubscription(opts ...controllertesting.SubscriptionOpt) *eventingv1alpha1.Subscription {
```

Comment:
I was wondering if this is needed (you can use the call within the function directly). But I think from a readability point of view, its good!

Reply:
Yes, as I explained to in yesterday this is needed for the readability purpose